chapter-4/e3-a-list.js

```diff
@@ -0,0 +1,59 @@
+function arrayToList(arr) {
+  return addValueToList(arr, list);
+
+  function addValueToList(arr, list) {
+    if (arr.length === 1) {
+      return {value: arr[0], rest: null};
+    }
+    let nextVal = arr.shift();
```

Comment:
it's worth noting that this will copy the entire remaining array over by one for each element in the array. You could do this with a direct index and it would have way better performance. Not every interview cares about that, but it's important to know.

Reply:
I'd also recommend practicing this